kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/TableCommitCoordinatorClientHandler.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.kernel.internal.snapshot;
+
+import java.io.IOException;
+import java.util.Map;
+
+import io.delta.kernel.data.Row;
+import io.delta.kernel.engine.CommitCoordinatorClientHandler;
+import io.delta.kernel.engine.coordinatedcommits.CommitFailedException;
+import io.delta.kernel.engine.coordinatedcommits.CommitResponse;
+import io.delta.kernel.engine.coordinatedcommits.GetCommitsResponse;
+import io.delta.kernel.engine.coordinatedcommits.UpdatedActions;
+import io.delta.kernel.utils.CloseableIterator;
+
+public class TableCommitCoordinatorClientHandler {
+    private final CommitCoordinatorClientHandler commitCoordinatorClientHandler;
+    private final String logPath;
+    private final Map<String, String> tableConf;
+
+    public TableCommitCoordinatorClientHandler(
+            CommitCoordinatorClientHandler commitCoordinatorClientHandler,
+            String logPath,
+            Map<String, String> tableConf) {
```

Comment:
@dhruvarya-db @prakharjain09 Why do we need to pass the `tableConf` to each API on the commit coordinator? Isn't it used when creating the commit coordinator?

Reply:
@vkorukanti Commit Coordinator is supposed to be reuseable across multiple tables (like JSONHandler). `tableConf` can be thought of as a `tablePath`/`tableId` which is used to uniquely identify the table.